src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -269,10 +269,35 @@ int sd_dhcp_client_set_mac(
                 const uint8_t *hw_addr,
                 const uint8_t *bcast_addr,
                 size_t addr_len,
-                uint16_t arp_type) {
+                unsigned short iftype) {
+
+        unsigned char default_eth_bcast[6] = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
+        unsigned char default_eth_hwaddr[6] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
```

Comment:
should be static const, and of type uint8_t?

Reply:
`const uint8_t` yes, but there's no need for static because we only need these buffers until `hw_addr_set` copies them into `client->hw_addr` and `client->bcast_addr`.

Reply:
well, immutable blobs of data should always be "static const" so that the linker can place it in a ro section in the ELF file.

const data on the stack otoh doesn't really make much sense, because you are basically telling the compiler "this data should not shipped in ELF, but generated at runtime on the stack, but afterwards it should be treated like ELF shipped data eh", which doesn't make too much sense.